client/state/invites/actions.js

```diff
@@ -219,33 +220,33 @@ export function disableInviteLinks( siteId ) {
 }
 
 export function acceptInvite( invite ) {
-	return ( dispatch ) => {
-		const result = wpcom.undocumented().acceptInvite( invite );
-		result
-			.then( ( data ) => {
-				if ( invite.role !== 'follower' && invite.role !== 'viewer' ) {
-					dispatch( receiveSites( data.sites ) );
-				}
-
-				if ( ! get( invite, 'site.is_vip' ) ) {
-					dispatch( successNotice( ...acceptedNotice( invite ) ) );
-				}
-
-				recordTracksEvent( 'calypso_invite_accepted', {
-					is_p2_site: get( invite, 'site.is_wpforteams_site', false ),
-					inviter_blog_id: get( invite, 'site.ID', false ),
-				} );
-
-				dispatch( inviteAccepted( invite ) );
-				dispatch( requestSites() );
-			} )
-			.catch( ( error ) => {
-				if ( error.message ) {
-					dispatch( errorNotice( error.message, { displayOnNextPage: true } ) );
-				}
-				recordTracksEvent( 'calypso_invite_accept_failed', { error: error.error } );
+	return async ( dispatch ) => {
+		const data = await wpcom.undocumented().acceptInvite( invite );
```

Comment:
This REST call happens outside the try/catch block, so the `catch` statement won't catch errors from it. I'll fix it in a PR that removes the `wpcom.undocumented()` methods.

Also, the original promise-based action returned a rejected promise in case of error. Now it returns a promise resolved to `undefined`.